content/articles/building-a-simple-nestjs-microservice-application-using-redis-as-the-transporter/index.md

```diff
@@ -0,0 +1,136 @@
+
+This tutorial will teach you how to create a Nest.js Microservice application with Redis.
+
+### An Overview of a microservice
+
+Microservice is an architectural design that allows you to create small modules/components of an application that can scale on demand. Each component is packaged independently and can communicate with each other via APIs. Nest.js comes with built-in support for microservice architecture and multiple transport layers. These transport layers are in charge of transferring messages between microservice instances. We can use Redis as a transport layer protocol for end-to-end connectivity to allow flow control and segmentation between different microservices. As a result, every microservice can connect to different channels, and every channel can have various microservices connected to it. Let’s now build a Nest.js microservice application using Redis as the transporter.
+
+### Prerequisites
+
+To follow along with this guide, the following prerequisites will be essential;
+
+- We are going to buddle Nest.js CLI using a Node.js package. Ensure you have Node.js installed so that you can have NPM on your computer. You can download the Node.js installer here.
+
+- Nest.js is bunddled and backed by Typescript. So a basic knowledge of Typescript is requried.
+
+- To implement a microservice architecture, we are going to use Redis as the transporter. Make sure Redis is installed on your computer.
+
+To build a Nest.js microservice, follow the below steps:
+
+### Step 1: Setting the application environments
+
+We will begin by installing the Nest.js CLI using the Node.js package manager. The Nest CLI is a command-line GUI framework that guides you in setting up, developing, and maintaining your Nest.js apps. It helps with many tasks, such as scaffolding a project, serving it in development mode, building and bundling the application for production distribution. The following command will help us install it globally and ready to interact with Nest.js
+
+```bash
+npm i -g @nestjs/cli
+```
+
+Now that we have Nest.js CLI ready, we can go ahead to build and create a Nest.js application template. Here is a command to set that up.
+
+```bash
+nest new nest-ms
+```
+
+We need to turn our newly created project into a monorepo. A monorepo refers to a single repository hosting different microservices. To generate the monorepo in the app, we run the following command.
+
+```bash
+nest generate app new_app
+```
```

Comment:
It's a bit unclear if I should run this command inside the nest-ms folder or outside. Please make it clear here.